crates/hotshot/types/src/data/vid_disperse.rs

```diff
@@ -591,3 +594,272 @@ impl<TYPES: NodeType> VidDisperseShare2<TYPES> {
+        self.epoch
+    }
+}
+
+impl<TYPES: NodeType> AvidmGf2DisperseShare<TYPES> {
+    /// Create a vector of `VidDisperseShare` from `VidDisperse`
+    pub fn from_vid_disperse(vid_disperse: AvidmGf2Disperse<TYPES>) -> Vec<Self> {
+        vid_disperse
+            .shares
+            .into_iter()
+            .map(|(recipient_key, share)| Self {
+                share,
+                recipient_key,
+                view_number: vid_disperse.view_number,
+                payload_commitment: vid_disperse.payload_commitment,
+                epoch: vid_disperse.epoch,
+                target_epoch: vid_disperse.target_epoch,
+                common: vid_disperse.common.clone(),
+            })
+            .collect()
+    }
+
+    /// Consume `self` and return a `Proposal`
+    pub fn to_proposal(
+        self,
+        private_key: &<TYPES::SignatureKey as SignatureKey>::PrivateKey,
+    ) -> Option<Proposal<TYPES, Self>> {
+        let Ok(signature) =
+            TYPES::SignatureKey::sign(private_key, self.payload_commitment.as_ref())
+        else {
+            tracing::error!("VID: failed to sign dispersal share payload");
+            return None;
+        };
+        Some(Proposal {
+            signature,
+            _pd: PhantomData,
+            data: self,
+        })
+    }
+
+    /// Create `VidDisperse` out of an iterator to `VidDisperseShare`s
+    pub fn to_vid_disperse<'a, I>(mut it: I) -> Option<AvidmGf2Disperse<TYPES>>
+    where
+        I: Iterator<Item = &'a Self>,
+    {
+        let first_vid_disperse_share = it.next()?.clone();
+        let payload_byte_len = first_vid_disperse_share.common.payload_byte_len();
+        let mut share_map = BTreeMap::new();
+        share_map.insert(
+            first_vid_disperse_share.recipient_key,
+            first_vid_disperse_share.share,
+        );
+        let mut vid_disperse = AvidmGf2Disperse {
+            view_number: first_vid_disperse_share.view_number,
+            epoch: first_vid_disperse_share.epoch,
+            target_epoch: first_vid_disperse_share.target_epoch,
+            payload_commitment: first_vid_disperse_share.payload_commitment,
+            shares: share_map,
+            payload_byte_len,
+            common: first_vid_disperse_share.common,
+        };
+        let _ = it.map(|vid_disperse_share| {
+            vid_disperse.shares.insert(
+                vid_disperse_share.recipient_key.clone(),
+                vid_disperse_share.share.clone(),
+            )
+        });
+        Some(vid_disperse)
+    }
+
+    /// Returns the payload length in bytes.
+    pub fn payload_byte_len(&self) -> u32 {
+        self.common.payload_byte_len() as u32
+    }
+
+    /// Split a VID share proposal into a proposal for each recipient.
+    pub fn to_vid_share_proposals(
+        vid_disperse: AvidmGf2Disperse<TYPES>,
+        signature: &<TYPES::SignatureKey as SignatureKey>::PureAssembledSignatureType,
+    ) -> Vec<Proposal<TYPES, Self>> {
```

Comment:
also feel strange to be a method on the Share rather than on `AvidmGf2Disperse.to_proposals(sig) -> Vec<Proposal>`